xarray/core/dataset.py

```diff
@@ -2674,7 +2755,12 @@ def set_index(self, indexes=None, append=False, inplace=None,
         return self._replace_vars_and_dims(variables, coord_names=coord_names,
                                            inplace=inplace)
 
-    def reset_index(self, dims_or_levels, drop=False, inplace=None):
+    def reset_index(
+        self,
+        dims_or_levels: Union[Hashable, Sequence[Hashable]],
+        drop: bool = False,
+        inplace: bool = None,
```

Comment:
`inplace` should go away soon so shouldn't matter... but should this be `Optional[bool]` given the default is `None`?

Reply:
See above